### ISSUE TITLE
STDLIB features

### ISSUE THREAD

**Issue by darkdarcool** [cursorweb/Europa-Lang#22]
# STDLIB features that need to be added

- [ ] io
  - [x] println
  - [x] print
  - [x] flush 
  - [x] readln
  - [ ] read (no new line)
  - [ ] passread (get input like read but no output when the user types it, produces no new line, but can if you wanna)
- [ ] Date
  - [ ] now
  - [ ] epoch
  - [ ] get_minute
  - [ ] get_second
  - [ ] get_day
  - [ ] get_month
  - [ ] get_year
- [ ] Path module like JS
  - [ ] join
- [ ] fs
  - [ ] read_file
  - [ ] write_file
  - [ ] file_exists
- [ ] http
  - [ ] get
  - [ ] post
  - [ ] put
  - [ ] all

**Comment by 19wintersp** [comment]
I think that kind of HTTP API is a bit too high-level, and you'd want a basic networking API before you even touch HTTP. An OS/env/system module is probably required, as would be the various type API's.

**Comment by darkdarcool** [comment]
Makes sense. You can edit my response or dm me on discord the things needed for the os. And I think an http api would be good as a separate cargo package. What do you think?

**Comment by cursorweb** [comment]
Yeah this is all planning in the distant future

**Comment by 19wintersp** [comment]
Sure, more complex API's could be shipped off as packages. For OS, I'd suggest looking at other Stdlibs like [Rust's](https://doc.rust-lang.org/std/#modules), Node's or Python's

**Comment by darkdarcool** [comment]
I like nodes straight up easy functions, like the type of os they have. But I also like Denos easy os/subprocess api. Refer to [this](https://deno.land/manual@v1.13.2/examples/subprocess)
